include/torch-mlir/Dialect/Torch/Utils/Utils.h

```diff
@@ -21,6 +21,10 @@ namespace Torch {
 int64_t toPositiveDim(int64_t dim, int64_t inputRank);
 bool isValidDim(int64_t dim, int64_t inputRank);
 bool getListConstructElements(Value v, SmallVectorImpl<Value> &elems);
+/// Returns the dimension indicated by `v` for a list of given `rank`.
```

Comment:
For consistency (and I also think it reads a little better), I think it would be nice to make this a matcher, so that instead of

```
        if (!matchPattern(setItem.idx(), m_TorchConstantInt(&index)))
          return failure();
```

We instead do
```
        if (!matchPattern(setItem.idx(), m_LegalConstantIndexIntoList(&index, runningList.size())))
          return failure();
```

Reply:
I think there is a separate discussion if we want to migrate some of these

```
        if (!matchPattern(value, m_Something(&thing)))
          return failure();
```

to

```
auto valueOpt = matchSomething(value);
if (!valueOpt)
  return failure();
```

or

```
int64_t thing;
if (matchSomething(&thing))
  return failure()
// (kind of like getListConstructElements)
```

as a matter of style. Right now I want to avoid having too many ways of matching in the codebase.

Reply:
Actually, on thinking about it more, I'll need to do a pass through the codebase to make this consistant anyways because of getListConstructElements, so for now, I'm fine moving forward with this patch, calling the function matchLegalConstantIndexIntoListOfSize. Sorry for all the back and forth.

Reply:
I've renamed the function as you suggested.

I understand the discussion on consistency. I'm happy to contribute to this. For `matchLegalConstantIndexIntoListOfSize` specifically there were more places in the codebase with the same tests, but with different messages produced for the different reasons it could fail. I think it would be useful to think if we'd want to keep the ability to generate different failure messages. I'd appreciate if this can indeed be done in a new PR.

Reply:
You might want to update the comment too -- it seems to be written from the perspective of a shape, but the utility is really about a general list.